pkg/tasks/prometheusoperator_user_workload.go

```diff
@@ -0,0 +1,176 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tasks
+
+import (
+	"github.com/openshift/cluster-monitoring-operator/pkg/client"
+	"github.com/openshift/cluster-monitoring-operator/pkg/manifests"
+
+	"github.com/pkg/errors"
+)
+
+type PrometheusOperatorUserWorkloadTask struct {
+	client  *client.Client
+	factory *manifests.Factory
+	config  *manifests.UserWorkloadConfig
+}
+
+func NewPrometheusOperatorUserWorkloadTask(client *client.Client, factory *manifests.Factory, config *manifests.UserWorkloadConfig) *PrometheusOperatorUserWorkloadTask {
+	return &PrometheusOperatorUserWorkloadTask{
+		client:  client,
+		factory: factory,
+		config:  config,
+	}
+}
+
+func (t *PrometheusOperatorUserWorkloadTask) Run() error {
+	if t.config.IsEnabled() {
+		return t.create()
+	}
+
+	return t.destroy()
+}
+
+func (t *PrometheusOperatorUserWorkloadTask) create() error {
+	sa, err := t.factory.PrometheusOperatorUserWorkloadServiceAccount()
+	if err != nil {
+		return errors.Wrap(err, "initializing UserWorkload Prometheus Operator ServiceAccount failed")
+	}
+
+	err = t.client.CreateOrUpdateServiceAccount(sa)
+	if err != nil {
+		return errors.Wrap(err, "reconciling UserWorkload Prometheus Operator ServiceAccount failed")
+	}
+
+	cr, err := t.factory.PrometheusOperatorUserWorkloadClusterRole()
+	if err != nil {
+		return errors.Wrap(err, "initializing UserWorkload Prometheus Operator ClusterRole failed")
+	}
+
+	err = t.client.CreateOrUpdateClusterRole(cr)
+	if err != nil {
+		return errors.Wrap(err, "reconciling UserWorkload Prometheus Operator ClusterRole failed")
+	}
+
+	crb, err := t.factory.PrometheusOperatorUserWorkloadClusterRoleBinding()
+	if err != nil {
+		return errors.Wrap(err, "initializing UserWorkload Prometheus Operator ClusterRoleBinding failed")
+	}
+
+	err = t.client.CreateOrUpdateClusterRoleBinding(crb)
+	if err != nil {
+		return errors.Wrap(err, "reconciling UserWorkload Prometheus Operator ClusterRoleBinding failed")
+	}
+
+	svc, err := t.factory.PrometheusOperatorUserWorkloadService()
+	if err != nil {
+		return errors.Wrap(err, "initializing UserWorkload Prometheus Operator Service failed")
+	}
+
+	err = t.client.CreateOrUpdateService(svc)
+	if err != nil {
+		return errors.Wrap(err, "reconciling UserWorkload Prometheus Operator Service failed")
+	}
+
+	denyNamespaces, err := t.client.NamespacesToMonitor()
+	if err != nil {
+		return errors.Wrap(err, "listing namespaces to monitor failed")
```

Comment:
Nit:
Hmm something like that:

```suggestion
		return errors.Wrap(err, "initializing userWorkload denied namespaces list failed")
```

Reply:
Especially since the only error we get from that is a wrapped `listing namespaces failed ` so it would be duplicated.